File: Models/LikeResult.cs
using PicturesAPI.Enums;
using PicturesAPI.Models.Dtos;

namespace PicturesAPI.Models;

public class LikeResult
{
    public LikeState LikeState { get; set; }
    public IEnumerable<LikeDto> Likes { get; set; }
    public int LikeCount { get; set; }
    public int DislikeCount { get; set; }
}

File: PooPosting/PooPosting.View/src/app/shared/utils/dtos/LikeResult.ts
import {LikeDto} from "./LikeDto";
import {LikeState} from "../enums/likeState";

export interface LikeResult {
  likeCount: number,
  dislikeCount: number,
  likes: LikeDto[],
  likeState: LikeState
}
